contentcuration/contentcuration/static/js/edit_channel/file_upload/views.js

```diff
@@ -594,12 +600,143 @@ var FormatSlot = BaseViews.BaseListNodeItemView.extend({
         });
     },
     remove_item:function(){
-        this.containing_list_view.set_file_format(null, this.model, this.file);
-        this.file = null;
-        this.render();
+        if(this.language_view){
+            this.language_view.add_language(new Models.LanguageModel(this.file.get('language')));
+            this.set_file(null);
+            this.remove();
+            this.containing_list_view.render();
+        }else{
+            this.set_file(null);
+            this.render();
+        }
+
     },
     set_uploading:function(uploading){
         this.containing_list_view.set_uploading(uploading);
+    },
+    load_subfiles:function(){
+        var data = {
+            node : this.node,
+            model: this.model,
+            el: $(this.getSubdirectory()),
```

Comment:
For situations like where we don't use any jquery methods, I'm not sure if it's better to use JQuery or `getElementById`. It's a pre-optimization though, so might not wanna even worry about it unless performance becomes an issue.